src/hoprd_secret.rs

```diff
@@ -98,10 +98,10 @@ impl SecretManager {
         Ok(secret)
     }
 
-    /// Gets the wallet secret linked to the hoprd-operator
+    /// Gets the Kubernetes secret used by the hoprd-operator to register and fund nodes 
     pub async fn get_wallet_secret(&self) -> Result<Secret, Error> {
         let api: Api<Secret> = Api::namespaced(self.client.clone(), &self.operator_config.instance.namespace);
-        if let Some(wallet_secret) = api.get_opt(&self.operator_config.instance.name).await? {
+        if let Some(wallet_secret) = api.get_opt(&self.operator_config.instance.secret_name).await? {
             Ok(wallet_secret)
         } else {
             Err(Error::SecretStatusError("[ERROR] Could not get wallet secret".to_owned()))
```

Comment:
This code should not be called. The `?` should short curcuit the error. Remove the `?` if this logic should be applied.

Reply:
I think that you already mentioned this issue before. The template is taken from the sample code of the get_opt function:

    /// [Get](`Api::get`) a named resource if it exists, returns [`None`] if it doesn't exist
    ///
    /// ```no_run
    /// # use kube::Api;
    /// use k8s_openapi::api::core::v1::Pod;
    ///
    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
    /// # let client: kube::Client = todo!();
    /// let pods: Api<Pod> = Api::namespaced(client, "apps");
    /// if let Some(pod) = pods.get_opt("blog").await? {
    ///     // Pod was found
    /// } else {
    ///     // Pod was not found
    /// }
    /// # Ok(())
    /// # }
    /// ```
    
    I'm just using the same pattern